### PR TITLE
feat: add url value component for use in re-designed confirmation pages

### LINKED ISSUE #11496: Create URL value component for key->value row
To be able to create personal sign page which a very basic confirmation page we need URL value component:

Screenshot 2024-09-30 at 4 19 50 PM

This should be similar to URL component in extension.

### PR DESCRIPTION
**Pull request by jpuri**
## **Description**

Add url value component for use in re-designed confirmation pages

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-mobile/issues/11496

## **Manual testing steps**

1. Run storybook locally
2. Check confirmations -> info row story

## **Screenshots/Recordings**
Screenshot 2024-10-02 at 3 08 32 PM

## **Pre-merge author checklist**

- [X] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [X] I've completed the PR template to the best of my ability
- [X] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [X] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by pedronfigueiredo — DISMISSED** [review]

File: app/components/Views/confirmations/components/UI/InfoRow/InfoValue/InfoURL/InfoURL.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  IconSize,
+} from '../../../../../../../../component-library/components/Icons/Icon';
+import { useTheme } from '../../../../../../../../util/theme';
+import createStyles from './style';
+
+interface InfoURLProps {
+  url: string;
+}
+
+const InfoURL = ({ url }: InfoURLProps) => {
```
Comment (matthewwalsh0) [inline app/components/Views/confirmations/components/UI/InfoRow/InfoValue/InfoURL/InfoURL.tsx:16]: Are we intentionally differing from the extension which calls this `ConfirmInfoRowUrl`?

File: app/components/Views/confirmations/components/UI/InfoRow/InfoValue/InfoURL/InfoURL.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  let urlObject;
+
+  try {
+    urlObject = new URL(url);
+  } catch (e) {
+    // eslint-disable-next-line no-console
+    console.log(`InfoURL: new URL(url) cannot parse ${url}`);
```
Comment (matthewwalsh0) [inline app/components/Views/confirmations/components/UI/InfoRow/InfoValue/InfoURL/InfoURL.tsx:23]: Should we be using `Logger` as we're in mobile?

**Review by matthewwalsh0 — DISMISSED** [review]

File: app/components/Views/confirmations/components/UI/InfoRow/InfoValue/InfoURL/InfoURL.tsx (hunk 1 above)
Comment (jpuri) [reply app/components/Views/confirmations/components/UI/InfoRow/InfoValue/InfoURL/InfoURL.tsx:16]: no, it is just bit more simplified I think

File: app/components/Views/confirmations/components/UI/InfoRow/InfoValue/InfoURL/InfoURL.tsx (hunk 2 above)
Comment (jpuri) [reply app/components/Views/confirmations/components/UI/InfoRow/InfoValue/InfoURL/InfoURL.tsx]: Done

File: CHANGELOG.md [hunk 3]
```diff
@@ -361,10 +361,7 @@
 - [#10383](https://github.com/MetaMask/metamask-mobile/pull/10383): fix: race condition issues when doing batch-rpc calls in the DeeplinkProtocolService (#10383)
 - [#10365](https://github.com/MetaMask/metamask-mobile/pull/10365): fix: attribution link (#10365)
 - [#10303](https://github.com/MetaMask/metamask-mobile/pull/10303): fix: page navigation during QR accounts selection (#10303)
-<<<<<<< HEAD
 
-=======
->>>>>>> main
 ## 7.28.1 - Aug 15, 2024
```
Comment (jpuri) [inline CHANGELOG.md:365]: Weird but these lined were part of main branch somehow.

**Review by matthewwalsh0 — APPROVED** [review]

**Pull request merged**